src/crab/ebpf_domain.cpp

```diff
@@ -1767,11 +1762,75 @@ void ebpf_domain_t::operator()(const Bin& bin) {
                         if (!bin.is64) {
                             input &= UINT32_MAX;
                         }
-                        uint64_t output = (int64_t)(input >> imm);
+                        // 0x8000000000000000 will throw an exception if passed to m_inv.set
+                        // as an int64_t interval per arsh64.data conformance test.
+                        // But mod64.data passes with an int64_t which passes 0xb185843600000000
+                        // as does stxdw.data which passes 0x8877665500000000, but that is because
+                        // they don't have relationships.  In the future we need a way of handling
+                        // relationships between the full range of uint64_t.
+                        uint64_t output = input << imm;
                         m_inv.set(dst.value, crab::interval_t{number_t{output}});
                         break;
                     }
+                } else if (interval.finite_size()) {
+                    number_t lb = interval.lb().number().value();
+                    number_t ub = interval.ub().number().value();
+                    if (bin.is64) {
+                        uint64_t lb_n = lb.cast_to_uint64() << imm;
+                        uint64_t ub_n = ub.cast_to_uint64() << imm;
+                        m_inv.set(dst.value, crab::interval_t{lb_n, ub_n});
```

Comment:
I prefer having `m_inv.set()` once, at the end of the outer branch. It is probably a matter of taste, but I think it'll make it more obvious how to extract a function such as `interval_t rshift_finite(interval_t)` at some point. In general, I think seperating interval manipulation from calls to SplitDBM methods is a good idea.

Reply:
Updated.  Can move it up another level in PR #459 which handles the first branch.